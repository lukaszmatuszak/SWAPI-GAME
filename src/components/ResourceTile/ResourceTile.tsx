import React from 'react';
import { Typography, Card, CardContent } from '@material-ui/core';
import { Resource } from '../../views/HomeView/resources';

const ResourceTile: React.FC<ResourceProps> = ({ resource }: ResourceProps) => (
    <Card>
        <CardContent>
            <img width="100%" src={resource.iconSrc} alt={resource.iconAlt} />
            <Typography align="center" variant="h3">
                {resource.title}
            </Typography>
        </CardContent>
    </Card>
);

interface ResourceProps {
    resource: Resource;
}

export default ResourceTile;
