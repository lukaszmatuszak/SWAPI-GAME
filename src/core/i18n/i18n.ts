import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

import locales from './locales/index';

i18n.use(initReactI18next)
    .use(LanguageDetector)
    .init({
        ns: ['Common'],
        defaultNS: 'Common',
        resources: locales,
        fallbackLng: 'en',
        interpolation: {
            escapeValue: false,
        },
    });
