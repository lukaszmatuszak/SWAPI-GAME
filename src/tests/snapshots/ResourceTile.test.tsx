import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ResourceTile from '../../components/ResourceTile/ResourceTile';
import '../../core/i18n/i18n';

describe('Spinner Snapshot - ', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <ResourceTile
                resource={{
                    iconSrc: '/vader.svg',
                    title: 'test',
                    iconAlt: 'testAlt',
                    redirect: 'test',
                }}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
