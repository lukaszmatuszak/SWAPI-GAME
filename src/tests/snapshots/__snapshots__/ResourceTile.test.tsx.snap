// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Snapshot -  renders correctly 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <img
          alt="testAlt"
          src="/vader.svg"
          width="80%"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Typography))
          variant="h3"
        >
          test
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
